buildSrc/src/main/kotlin/UpdateEmojiValuesTask.kt

```diff
@@ -0,0 +1,127 @@
+    abstract val targetPackageName: Property<String>
+
+    @TaskAction
+    fun updateEmojiValues() {
+        val urlContents = URL(emojiSequencesUrl.get()).readText()
+
+        val basicValues = mutableListOf<Int>()
+        val basicRanges = mutableListOf<IntRange>()
+        val valuePairs = mutableListOf<Pair<Int, Int>>()
+        val valueSequences = mutableListOf<List<Int>>()
+        for (line in urlContents.lineSequence()) {
+            val spec = line.substringBefore(";", missingDelimiterValue = "").trim()
+            when {
+                spec.isEmpty() -> continue
+                ".." in spec -> {
+                    val (start, end) = spec.split("..").map { it.toInt(16) }
+                    basicRanges.add(start..end)
+                }
+                else -> {
+                    val values = try {
+                        spec.split(" ").map { it.toInt(16) }
+                    } catch (e: NumberFormatException) {
+                        continue
+                    }
+                    when (values.size) {
+                        1 -> basicValues.add(values[0])
+                        2 -> {
+                            val high = values[0]
+                            if ((high !in basicValues) && basicRanges.none { high in it })
+                                valuePairs.add(Pair(values[0], values[1]))
+                        }
+                        else -> valueSequences.add(values)
+                    }
+                }
+            }
+        }
+
+        val indent = "    "
+        File(targetDirectory.get().asFile, "EmojiValues.kt").bufferedWriter(Charsets.UTF_8).use { output ->
+            output.appendLine("package ${targetPackageName.get()}")
+            output.appendLine()
+
+            val date =  LocalDate.now().format(DateTimeFormatter.ofPattern("yyyy-MM-dd"))
+            output.appendLine("// Generated from ${emojiSequencesUrl.get()} on $date")
+
+            output.appendLine("internal val EMOJI_RANGES = arrayOf(")
+            for (r in basicRanges) {
+                output.append(indent)
+                output.appendLine("EmojiRange(0x${r.first.toString(16)}..0x${r.last.toString(16)}),")
+            }
+            output.appendLine(")")
+            output.appendLine()
+
+            output.appendLine("internal val EMOJI_BASIC_VALUES = hashSetOf(")
+            for (v in basicValues) {
+                output.append(indent)
+                output.appendLine("EmojiBasicValue(0x${v.toString(16)}),")
+            }
+            output.appendLine(")")
+            output.appendLine()
+
+            output.appendLine("// Note that this excludes pairs where the first value is already in EMOJI_RANGES or EMOJI_BASIC_VALUES")
+            output.appendLine("internal val EMOJI_VALUE_PAIRS = hashSetOf(")
+            for (v in valuePairs) {
+                output.append(indent)
+                output.appendLine("EmojiValuePair(0x${v.first.toString(16)}, 0x${v.second.toString(16)}),")
```

Comment:
It uses internal foundation's `EmojiValuePair` type, so it doesn't really usable outside of that module. So, placing it to `buildSrc` is quite questionable

Reply:
It needs to be accessible from the gradle task. Should I put it in foundation/buildSrc?

Reply:
`UpdateTranslationsTask.kt` is right there beside `UpdateEmojiValuesTask.kt`.

I can't put Kotlin code inside `.gradle` (it's Groovy, not kts), and I don't know how to set up `foundation/foundation/buildSrc` correctly.